app/src/main/java/com/github/warnastrophy/core/data/repository/HazardsRepository.kt

```diff
@@ -52,61 +81,55 @@ class HazardsRepository() : HazardsDataSource {
    * @param urlStr The URL to fetch.
    * @return The response body as a String, or an empty string if the connection fails.
    */
-  private fun httpGet(urlStr: String): String =
-      with(Dispatchers.IO) {
-        val url = URL(urlStr)
-        val conn =
-            (url.openConnection() as HttpURLConnection).apply {
-              requestMethod = "GET"
-              setRequestProperty("Accept", "application/json")
-              connectTimeout = 15000
-              readTimeout = 15000
-            }
-        try {
-          val code = conn.responseCode
-          val stream = if (code in 200..299) conn.inputStream else conn.errorStream
-          BufferedReader(InputStreamReader(stream)).use { it.readText() }
-        } finally {
-          conn.disconnect()
+  private suspend fun httpGet(urlStr: String): String {
```

Comment:
```suggestion
  private suspend fun httpGet(urlStr: String): String = withContext(Dispatchers.IO) {
      // httpGet body
  }
```
Even if `httpGet` is now a `suspend` function, if it is called from a Main/Default coroutine context, the blocking `HttpURLConnection` I/O will block that dispatcher (UI/main thread).

Notes:
- `suspend` enables pausing and resuming, but does not itself manage threads.
- `withContext(Dispatchers.IO)` explicitly switches to a background thread for blocking I/O.